src/compiler/program.ts

```diff
@@ -1180,6 +1180,31 @@ namespace ts {
 
             return result;
 
+            function doesStructuralPermitResolutionsReuse(): boolean {
+                switch (structuralIsReused) {
+                    case StructureIsReused.Not: return false;
+                    case StructureIsReused.SafeProjectReferenceModules:
+                        return file.resolvedModules !== undefined &&
+                            isSourceOfProjectReferenceRedirect(file.originalFileName);
+                    case StructureIsReused.SafeModules: return true;
+                    case StructureIsReused.Completely: return true;
+
+                    // It's possible for resolveModuleNamesReusingOldState to be called by
+                    // tryReuseStructureFromOldProgram before structuralIsReused has been defined.
+                    // In this case, the caller (tryReuseStructureFromOldProgram) expects this
+                    // function to reuse resolutions for its checks. Returning true to accommodate
+                    // for that specific scenario.
+                    //
+                    // TODO: Clean this because the above scenario is a bit unexpected. This case
+                    // should ideally return false or throw.
+                    case undefined: return true;
+
+                    // The above cases should exhaustively cover all branches. Defaulting to false
+                    // since that's safer in unexpected situations.
+                    default: return false;

```

Comment:
This isn't an exhaustive switch statement because `StructureIsReused` sets bit flags manually. (I assume for performance reasons.) Is it worth keeping it that way?